### PR TITLE
Suppress logging during dropwizard-validations tests

### PR DESCRIPTION
**Pull request by rhowe**
Suppress console logging during these tests

### CODE REVIEW

File: dropwizard-validation/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <outputPatternAsHeader>false</outputPatternAsHeader>
```
Comment (zUniQueX) [inline dropwizard-validation/src/test/resources/logback-test.xml]: This feature is [documented](https://logback.qos.ch/manual/encoders.html#outputPatternAsHeader) as disabled by default. Any reason this line must be added?

**Comment by zUniQueX** [comment]
Thanks for the contribution!

I don't think it's a good idea to have different levels in the `logback-test.xml` and the filtering of the `ListAppender`. This PR would change the semantics of the `isEmpty()` check on the list of the `ListAppender`. Wouldn't it be a better idea to call `setAdditive(false)` for the logger of `SelfValidationTest` and use only the `ListAppender` and not the console appender?

As we do not rely on logging in the other tests, I don't have a strong opinion, if it's a good idea to set the log event globally or locally. But I'd want to avoid a situation, where the semantics of the appender would confuse developers.

File: dropwizard-validation/src/test/resources/logback-test.xml (hunk 1 above)
Comment (joschi) [reply dropwizard-validation/src/test/resources/logback-test.xml]: Probably copy & pasted from some other source. Feel free to remove this line.

**Comment by rhowe** [comment]
@zUniQueX I agree with everything you say, I just didn't see a nice alternative. I'll have a play around and see.

**Comment by rhowe** [comment]
> @zUniQueX I agree with everything you say, I just didn't see a nice alternative. I'll have a play around and see.

Alright, I think I've managed to do something not too ugly...

File: dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java [hunk 2]
```diff
@@ -55,10 +58,12 @@ protected void append(ILoggingEvent iLoggingEvent) {
     @BeforeAll
     static void setUp() {
         LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
-        Logger root = loggerContext.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
-        listAppender.setContext(root.getLoggerContext());
+        Logger selfValidatingValidatorLogger = loggerContext.getLogger(SelfValidatingValidator.class);
+        selfValidatingValidatorLogger.setAdditive(false);
+        selfValidatingValidatorLogger.setLevel(Level.DEBUG);
```
Comment (zUniQueX) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java]: Logback defines the `DEBUG` level by default, but our appender can only process events with level `INFO` or higher. We _could_ change the level, but I don't have a strong opinion on either version.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java [hunk 3]
```diff
@@ -360,6 +368,13 @@ void violationMessagesAreEscapedByDefault() {
 
     @Test
     void messageParametersExample() {
+        Logger hibernateLogger = (Logger)LoggerFactory.getLogger(AbstractMessageInterpolator.class);
+        hibernateLogger.setAdditive(false);
+        hibernateLogger.setLevel(Level.DEBUG);
```
Comment (zUniQueX) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java]: Same as above with the `INFO` level.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java [hunk 4]
```diff
@@ -55,10 +58,12 @@ protected void append(ILoggingEvent iLoggingEvent) {
     @BeforeAll
     static void setUp() {
         LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
```
Comment (zUniQueX) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java]: We don't need the `LoggerContext` here. As you already did in the `messageParametersExample()` method, we can get Logback's `Logger` object by casting the logger obtained from the `LoggerFactory.getLogger(...)` call.

File: dropwizard-validation/src/test/resources/logback-test.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+    <root level="off">
+        <appender-ref ref="STDOUT"/>
```
Comment (zUniQueX) [inline dropwizard-validation/src/test/resources/logback-test.xml:8]: I wonder, if we need to define an `<appender-ref />` here. If we turn off logging, we don't need an appender IMO. The tests work, if I remove the appender so maybe we don't need it.

**Review by zUniQueX — APPROVED** [review]
Thanks, LGTM!

I've just marked a few minor things you _could_ change, but don't have to. Feel free to change or ignore them.

File: dropwizard-validation/src/test/resources/logback-test.xml (hunk 5 above)
Comment (rhowe) [reply dropwizard-validation/src/test/resources/logback-test.xml:8]: Hm, good question. This is the same `logback-test.xml` we have everywhere else. I'll raise a separate PR removing that from all of them and we can discuss the merits there.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java (hunk 2 above)
Comment (rhowe) [reply dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java]: No strong opinions either - as you deduced I was picking up on the logback default, but maybe we can shave some microseconds off the test run by upping the log level to INFO. I'll do that.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java (hunk 4 above)
Comment (rhowe) [reply dropwizard-validation/src/test/java/io/dropwizard/validation/SelfValidationTest.java]: Right you are, it's gone.

**Comment by zUniQueX** [comment]
Thanks for the changes :+1:

**Pull request merged**